pex/third_party/__init__.py

```diff
@@ -116,10 +120,11 @@ def iter_root_packages(self, relpath):
       yield package
 
   def _filter_names(self, relpath, pattern, group):
-    pat = re.compile(r'^{prefix}{pattern}$'
-                     .format(prefix=self.prefix + ((relpath + os.sep) if relpath else ''),
-                             pattern=pattern))
-
+    # We use '/' here instead of os.sep because the zip file format spec specifies that paths must
```

Comment:
With all these suggestions, you'll also want to update both the comments to say `"We replace os.sep with '/' because..."`.